code/__HELPERS/unsorted.dm

```diff
@@ -1396,3 +1396,16 @@ GLOBAL_DATUM_INIT(dview_mob, /mob/dview, new)
 //checks if a turf is in the planet z list.
 /proc/turf_z_is_planet(turf/T)
 	return GLOB.z_is_planet["[T.z]"]
+
+// \ref behaviour got changed in 512 so this is necesary to replicate old behaviour.
+// If it ever becomes necesary to get a more performant REF(), this lies here in wait
+// #define REF(thing) (thing && istype(thing, /datum) && thing:use_tag && thing:tag ? "[thing:tag]" : "\ref[thing]")
+
+/proc/REF(input)
+	if(istype(input, /datum))
```

Comment:
better off using a typecache methinks

Reply:
Probably yeah, that's a massive typecache though.

Reply:
no, type caches only work if you can assume it is a datum, but \ref has to work properly on strings and numbers and lists for certain things

Reply:
I just took a shit and realized that but you ninja'd me >:(

Reply:
Ba dum tsh



Now I gotta shit too